usefulUtilities/voteApp/appResources/voteApp.js

```diff
@@ -0,0 +1,1022 @@
+                if (request.readyState === REQUEST_READY_STATE) {
+                    onCompleteCallback(request);
+                }
+            };
+            request.send();
+        },
+
+        loggedIn: {
+
+            handleNotLoggedInStatus: function () {
+
+                if (!setupNotLoggedIn) {
+                    if (DEBUG) {
+                        print("handleNotLoggedInStatus", setupNotLoggedIn);
+                    }
+
+                    dataStore.loggedin = false;
+                    AccountServices.loggedInChanged.connect(loggedIn);
+                    setupNotLoggedIn = true;
+                    dataStore.loading = false;
+                    updateUI();
+                }
+
+                function loggedIn() {
+                    AccountServices.loggedInChanged.disconnect(loggedIn);
+                    setupNotLoggedIn = false;
+                    dataStore.loggedin = true;
+                }
+            }
+
+        },
+
+        unloadByDate: {
+
+            willUnload: function () {
+
+                if (DEBUG) {
+                    print("UnloadByDate willUnload() check date");
+                }
+                var now = new Date();
+
+                if (now > UNLOAD_DATE) {
+
+                    var scriptList = ScriptDiscoveryService.getRunning();
+                    var url;
+                    scriptList.forEach(function (scriptInfo) {
+                        if (scriptInfo.name === VOTE_APP_NAME) {
+                            url = scriptInfo.url;
+                            dataStore.unload = true;
+                            dataStore.loading = false;
+
+                            updateUI();
+                        }
+                    });
+
+                    if (DEBUG) {
+                        print("UnloadByDate TRUE past UNLOAD_DATE");
+                    }
+                    Script.setTimeout(function () {
+
+                        ScriptDiscoveryService.stopScript(url);
+
+                    }, UNLOAD_TIME);
+
+                    return true;
+                }
+
+                return false;
+            }
+
+        },
+
+        loading: {
+            start: function () {
+                dataStore.loading = true;
+                updateUI();
+            },
+
+            end: function () {
+                dataStore.loading = false;
+                updateUI();
+            }
+        },
+
+        hasUserVoted: function(type) {
+
+            var voteAppSettings = this.getVoteAppSettings();
+    
+            if (DEBUG) {
+                print("Settings says:", voteAppSettings.voted[type], voteAppSettings.voted[type] !== "");
+            }
+    
+            return voteAppSettings.voted[type] !== "";
+        },
+
+        getVoteAppSettings: function() {
+
+            var currentSettings = Settings.getValue(VOTE_APP_SETTINGS_NAME); // , DEFAULT_VOTE_APPSETTINGS);
```

Comment:
I'm confused about why you're _not_ using the code that's commented out here. could you please help me understand?